streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamKStreamJoinTest.java

```diff
@@ -156,6 +156,69 @@ public void testJoin() throws Exception {
         processor.checkAndClearProcessResult("0:X0+YYY0", "0:X0+YYY0", "0:XX0+YYY0", "1:X1+YYY1", "1:X1+YYY1", "1:XX1+YYY1");
     }
 
+
+    @Test
+    public void testJoinWithKey() throws Exception {
```

Comment:
These tests that have been added are largely the same as the non key tests. Is it possible to refactor them so that they share the common code?

Reply:
@dguy I tried to apply tests (with and without key) through a common method (like `testJoin(boolean withKey)`), however, I got Cyclomatic complexity error while testing. As a result, I decided to pass some complexity to another function `KStreamKstreamJoinTest.checkResult()`.